### DISCUSSION TITLE
`Device` is Kind or Type?

### DISCUSSION THREAD

**Discussion by wangkanai** [wangkanai/wangkanai#315] [Ideas]
`Device` is kind of complicated. Because in English speaking, we would normally say;

* What **Type** of device is this?
* What **Kind** of device is this?

So should I make a breaking change by rename it to `Kind` or leave it unchange as `Type`?

_Originally posted by @wangkanai in https://github.com/wangkanai/Detection/pull/314#issuecomment-582902925_

```html
<ul>
    <li>@Detection.Device.Type</li>
    <li>@Detection.Browser.Name</li>
    <li>@Detection.Platform.Name</li>
    <li>@Detection.Engine.Name</li>
</ul>
```

**Comment by wangkanai** [comment]
Nobody got any comments, ideas, or feedback on this breaking change?

**Comment by frankhaugen** [comment]
@wangkanai "variant" is also a synonym.

But having English as a second language, my intuition is that "type" is the correct term.

"Kind" is for me first associated with "kindness". Don't change it, as it's a needless breaking change, that will not improve anything

**Comment by wangkanai** [comment]
@frankhaugen Thank you very much for the input. Much appreciated :)

I don't remember from the back of my head that I change device from `Type` to `Kind` or not. English is also my second language too, so i not which is best.

Another issue I have with this issue in the context of C# language, is the word `Type` is much associate with `GetType()` of [System Object Class]( https://docs.microsoft.com/en-us/dotnet/api/system.object.gettype?view=netcore-3.1) and [System Type Class](https://docs.microsoft.com/en-us/dotnet/api/system.type?view=netcore-3.1). This make it confusing for me is it the `Type` of object or `Type` of `Device`.

> "Kind" is for me first associated with "kindness". Don't change it, as it's a needless breaking change, that will not improve anything

Yap, it's a breaking change. So I'm would like to have a lot of input on this and maybe vote (wish too much)

**Comment by frankhaugen** [comment]
My colleague, (who is from England, and speaks English natively), "votes" for "Type", because in it's context it's obvious that it's not something related to C#. He would not call it neither "kind" nor "variant" as it's less precise the "Type". Suggest it's kept as "Type" and if someone raises an issue seeing it as ambiguous, it can be re-evaluated

**Comment by wangkanai** [comment]
Yap, I agree with you. If the issue become too ambiguous. We will re-evaluate the issue again. For now, let leave the issue as is.

**Comment by wangkanai** [comment]
move to discussion #474
